feat: default values for form elements (#14289)
sveltejs/svelte: documentation/docs/03-template-syntax/11-bind.md

If the input is empty or invalid (in the case of `type="number"`), the value is `undefined`.

> You can give the input a default value by setting the `defaultValue` property. This way, when the input is part of a form and its `form.reset()` method is invoked, it will revert to that value instead of the empty string. Note that for the initial render the value of the binding takes precedence if it's not `null` or `undefined`.

Reviewer: I think it probably makes sense to talk more abstractly about form resets, since in the provided example we're not using the `form.reset()` method (at least, not explicitly)

```suggestion
If an `<input>` has a `defaultValue` and is part of a form, it will revert to that value instead of the empty string when the form is reset. Note that for the initial render the value of the binding takes precedence unless it is `null` or `undefined`.
```

```svelte
<script>
	let value = $state('');
</script>

<form>

> 	<input bind:value defaultValue="x">

Reviewer: ```suggestion
	<input bind:value defaultValue="not the empty string">
```

	<input type="reset" value="Reset">
</form>

> ```

Reviewer: MDN [has a note about default values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/reset) which I think it worth reiterating here

````suggestion
```

> [!NOTE]
> Use reset buttons sparingly, and ensure that users won't accidentally click them while trying to submit the form.
````

</label>
```

> You can give the input a default value by setting the `defaultChecked` property. This way, when the input is part of a form and its `form.reset()` method is invoked, it will revert to that value instead of `false`. Note that for the initial render the value of the binding takes precedence if it's not `null` or `undefined`.

Reviewer: ```suggestion
If an `<input>` has a `defaultChecked` attribute and is part of a form, it will revert to that value instead of `false` when the form is reset. Note that for the initial render the value of the binding takes precedence unless it is `null` or `undefined`.
```

</select>
```

> You can give the select a default value by setting the `selected` property on the elements that should be selected initially. This way, when the select is part of a form and its `form.reset()` method is invoked, it will revert to that value instead of the empty string (for single-value selects) or the empty array (for multi-value selects). Note that for the initial render the value of the binding takes precedence if it's not `undefined`.

Reviewer: 'instead of the empty string' doesn't really make sense here, because the default behaviour if no option has the `selected` attribute is to select the first option. I think we can just leave that implied

```suggestion
You can give the `<select>` a default value by adding a `selected` attribute to the`<option>` (or options, in the case of `<select multiple>`) that should be initially selected. If the `<select>` is part of a form, it will revert to that selection when the form is reset. Note that for the initial render the value of the binding takes precedence if it's not `undefined`.
```
